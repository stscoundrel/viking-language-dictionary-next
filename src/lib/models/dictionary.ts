import { DictionaryEntry as RawDictionaryEntry } from 'viking-language-dictionary/dist/models'

export type OriginalDictionaryEntry = RawDictionaryEntry

export interface DictionaryEntry extends RawDictionaryEntry {
    slug: string,
}
