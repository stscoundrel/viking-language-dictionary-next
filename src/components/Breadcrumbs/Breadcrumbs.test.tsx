import { ContentType } from 'lib/models/content-types'
import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import Breadcrumbs from './index'

describe('Breadcrumbs component', () => {
  const word = {
    headword: 'þess',
    definition: 'm/n gen sg of sá/þat',
    isCommon: false,
    slug: 'thess',
  }

  test('Default page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs type={ContentType.Page} content={null} />)
  })

  test('Default page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs type={ContentType.Page} content={null} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Default page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs type={ContentType.Page} content={null} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(1)
  })

  test('Letter page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs type={ContentType.Letter} content={[word]} />)
  })

  test('Letter page: matches snapshot', () => {
    const tree = renderer.create(
      <Breadcrumbs type={ContentType.Letter} content={[word]} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Letter page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs type={ContentType.Letter} content={[word]} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(2)
  })

  test('Word page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs type={ContentType.Word} content={word} />)
  })

  test('Word page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs type={ContentType.Word} content={word} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Word page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs type={ContentType.Word} content={word} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(3)
  })
})
