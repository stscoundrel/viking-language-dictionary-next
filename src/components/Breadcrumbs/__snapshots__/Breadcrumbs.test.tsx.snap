// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component Default page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Viking Language Dictionary
    </a>
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"@context\\":\\"https://schema.org/\\",\\"@type\\":\\"BreadcrumbList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"name\\":\\"Viking Language Dictionary\\",\\"item\\":\\"undefined/\\"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;

exports[`Breadcrumbs component Letter page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Viking Language Dictionary
    </a>
    <a
      className="link"
      href="/letter/th"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Letter Þ
    </a>
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"@context\\":\\"https://schema.org/\\",\\"@type\\":\\"BreadcrumbList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"name\\":\\"Viking Language Dictionary\\",\\"item\\":\\"undefined/\\"},{\\"@type\\":\\"ListItem\\",\\"position\\":2,\\"name\\":\\"Letter Þ\\",\\"item\\":\\"undefined/letter/th\\"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;

exports[`Breadcrumbs component Word page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Viking Language Dictionary
    </a>
    <a
      className="link"
      href="/letter/th"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Letter Þ
    </a>
    <a
      className="link"
      href="/word/thess"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Þess
    </a>
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"@context\\":\\"https://schema.org/\\",\\"@type\\":\\"BreadcrumbList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"name\\":\\"Viking Language Dictionary\\",\\"item\\":\\"undefined/\\"},{\\"@type\\":\\"ListItem\\",\\"position\\":2,\\"name\\":\\"Letter Þ\\",\\"item\\":\\"undefined/letter/th\\"},{\\"@type\\":\\"ListItem\\",\\"position\\":3,\\"name\\":\\"Þess\\",\\"item\\":\\"undefined/word/thess\\"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;
